feat: Add `-chainlocknotify` cmd-line option, update `-instantsendnotify`

UdjinM6 opened the pull request:
## Issue being fixed or feature implemented
Execute command when the best chainlock changes (`%s` in cmd is replaced by chainlocked block hash). Same as `-blocknotify` but for chainlocks. Let `-instantsendnotify` replace `%w` with wallet name like `-walletnotify` does.

## What was done?

## How Has This Been Tested?
run tests

## Breaking Changes
n/a

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

knst reviewed (commented):
why do we need to create a brand new functional test with mostly copy-paste code?

It seems as possible to extend older test with manageable amount of efforts, or do I miss some important difference?

UdjinM6 commented:
> why do we need to create a brand new functional test with mostly copy-paste code?

I wouldn't agree with `mostly copy-paste code` part. I only borrowed like 3 variables and 1 function. :)

> It seems as possible to extend old test `test/functional/feature_notifications.py` with manageable amount of efforts, or do I miss some important difference?

That's how it would look like [d1461b0](https://github.com/dashpay/dash/pull/5522/commits/d1461b00efb8fe547b6da2ac087bbebd771a3863). Had to implement a few workarounds to make it work, will most likely cause merge conflicts in the future.

knst reviewed (commented):
> That's how it would look like [d1461b0](https://github.com/dashpay/dash/pull/5522/commits/d1461b00efb8fe547b6da2ac087bbebd771a3863). Had to implement a few workarounds to make it work,

yes, indeed, I counted 3 of them:
 - `self.nodes[1]` -> `rpc`
 - different `extra_args` logic (because we have non-empty args by default)
 - different way to create wallets & removing files

Despite that I think it is still better that own file for feature_notifications_dash.py by 2 reasons:
 - conflicts doesn't look as really difficult resolve in this particular case because the overall architecture and logic is not changed: all changes are in details of implementation
 - if it would be separate script `feature_nofications_dash.py` then no any changes from original `feature_notifications.py` would be ever backported to `feature_nofications_dash.py` -> over time difference between these scripts would be bigger and bigger and potentially it increase complexity of support over time.

Anyway, I am not strongly against of separate script, but I'd prefer having them together.

-------

Btw, why do we need to mine 4 quorums here? It works even with one quorum (as I tested locally). Otherwise LGTM now.

UdjinM6 commented:
Dropped `feature_nofications_dash.py`, squashed all fixes. Also added e7236f0 which should make diff smaller by avoiding "`self.nodes[1]` -> `rpc`" part.

knst reviewed (approved):
utACK

thephez reviewed (approved):
Slightly-tested ACK (ran with `...  -chainlocknotify="notify-send New ChainLock"`)


![image](https://github.com/dashpay/dash/assets/8145677/61e2f81e-9b34-4baf-bde9-5a517d2539c1)
